ui/app/components/ui/tooltip.js

```diff
@@ -1,6 +1,171 @@
+    this.tooltipEl.style.top = tooltipPosition.y + tooltipOffset.y + 'px'
+    this.tooltipEl.style.opacity = 1
+  };
+
+  handleMouseOut = () => {
+    this.resetTooltip()
+  };
+
+  getTooltipPosition = (e) => {
+    let pointX
+    let pointY
+    const bodyRect = document.body.getBoundingClientRect()
+    const containerRect = this.container.getBoundingClientRect()
+    const containerOffsetX = containerRect.left - bodyRect.left
+    const containerOffsetY = containerRect.top - bodyRect.top
+    if (this.props.fixed) {
+      const componentRect = this.componentEl.getBoundingClientRect()
+      const componentOffsetX = componentRect.left - containerOffsetX
+      const componentOffsetY = componentRect.top - containerOffsetY
+      const componentWidth = this.componentEl.offsetWidth
+      const componentHeight = this.componentEl.offsetHeight
+      let cOffsetX = 0
+      let cOffsetY = 0
+      switch (this.props.position) {
+        case 'top':
+          cOffsetX = componentWidth / 2
+          cOffsetY = 0
+          break
+        case 'right':
+          cOffsetX = componentWidth
+          cOffsetY = componentHeight / 2
+          break
+        case 'bottom':
+          cOffsetX = componentWidth / 2
+          cOffsetY = componentHeight
+          break
+        case 'left':
+          cOffsetX = 0
+          cOffsetY = componentHeight / 2
+          break
+        default:
+          cOffsetX = componentWidth / 2
+          cOffsetY = 0
+          break
+      }
+      pointX = componentOffsetX + cOffsetX + (window.scrollX || window.pageXOffset)
+      pointY = componentOffsetY + cOffsetY + (window.scrollY || window.pageYOffset)
+    } else {
+      const clientX = e.clientX
+      const clientY = e.clientY
+      pointX = clientX - containerOffsetX + (window.scrollX || window.pageXOffset)
+      pointY = clientY - containerOffsetY + (window.scrollY || window.pageYOffset)
+    }
+    return {
+      x: pointX,
+      y: pointY,
+    }
+  };
+
+  getTooltipOffset = () => {
+    const tooltipW = this.tooltipEl.offsetWidth
+    const tooltipH = this.tooltipEl.offsetHeight
+    let offsetX = 0
+    let offsetY = 0
+    switch (this.props.position) {
+      case 'top':
+        offsetX = -(tooltipW / 2)
+        offsetY = -(tooltipH + Number(this.props.space))
+        break
+      case 'right':
+        offsetX = Number(this.props.space)
+        offsetY = -(tooltipH / 2)
+        break
+      case 'bottom':
+        offsetX = -(tooltipW / 2)
+        offsetY = Number(this.props.space)
+        break
+      case 'left':
+        offsetX = -(tooltipW + Number(this.props.space))
+        offsetY = -(tooltipH / 2)
+        break
+      default:
+        offsetX = -(tooltipW + Number(this.props.space))
+        offsetY = -(tooltipH / 2)
+        break
+    }
+    return {
+      x: offsetX,
+      y: offsetY,
+    }
+  };
+
+  render () {
+    return this.props.children
```

Comment:
I think I know why swapping out this tooltip for `tooltip-v2` seems to break things: `tooltip-v2` adds a wrapper div, which breaks styles.

Both of these tooltips are terrible :disappointed: . We should replace both with `react-popper`, which is the "official" React wrapper for `popper.js` (the same library that `react-tippy` uses to calculate position).